doc/api/packages.md

```diff
@@ -357,6 +378,50 @@ import submodule from 'es-module-package/private-module.js';
 // Throws ERR_PACKAGE_PATH_NOT_EXPORTED
 ```
 
+#### Extensioned v Extensionless Subpaths
+
+It is recommended to pick either one of providing extensioned or unextensioned
+subpaths for a package to ensure consistent usage. This keeps the package
+contract clear for consumers, simplifies package subpath completions, and
+ensures all dependent packages import with the same specifier mappings.
+
+Ultimately, it is the choice of the package author which form of subpath to
+support - extensionless, like `import 'pkg/subpath'`, or extensioned, like
+`import 'pkg/subpath.js'` per the exports example above. Both conventions are
+used in the Node.js ecosystem.
+
+For packages where interoperability with [import maps][] is desired, using
```

Comment:
All package authors should desire interoperability with import maps. Even if Node never supports the feature (but it will) there are still things like Deno's Node compatibility mode. We should encourage whatever's the best approach for compatibility, both with the other runtimes of today and with Node's likely future.

Reply:
I think that's why this phrasing is inaccurate - extensionless and extensioned are both identically interoperable with import maps. What package authors are likely to not all prioritize is minimizing the size of an import map, since "ergonomics" and "aesthetics" are often valued more than "bytes over the wire".